### PR TITLE
Add config: 'no_data_partition'

### PR DESCRIPTION
**Pull request by johnliu55tw**
- Add doc for `no_data_partition` config.

- Remove `force_mbr` from `os` section.

- Reference `no_data_partition` from `force_mbr` config.

### CODE REVIEW

File: docs/install/harvester-configuration.md [hunk 1]
```diff
@@ -441,3 +442,18 @@ install:
   vip_mode: dhcp
   vip_hw_addr: 52:54:00:ec:0e:0b
 ```
+
+### `install.no_data_partition`
+
+#### Definition
+
+Do not create an additional disk partition for storing VM data
+and use operating system partition to store them. Default: `false`.
```
Comment (nunix) [inline docs/install/harvester-configuration.md:451]:
```suggestion
Do not create an additional disk partition for storing VM data. The operating system partition will be used to store the VM data. 

Default: `false`.
```

Comment (guangbochen) [inline docs/install/harvester-configuration.md:451]: Do not create additional disk partitions for VM data, the system partition will be used by default.
Can't be resized or reverted after the installation, useful when users want to use additional disks for the VM data with [auto-provision-path]() config. Default: `false`.

**Review by nunix — COMMENTED** [review]
rephrased the description

File: docs/install/harvester-configuration.md (hunk 1 above)
Comment (johnliu55tw) [reply docs/install/harvester-configuration.md]: Rephrased it a bit based on your suggestion.

Comment (johnliu55tw) [reply docs/install/harvester-configuration.md]: Rephrased it a bit based on your suggestion.

**Comment by johnliu55tw** [comment]
@guangbochen @nunix I think we need to merge this PR first because I've fixed some minor issues, then I will rebase my other PRs based on this.

**Review by nunix — APPROVED** [review]

**Review by guangbochen — APPROVED** [review]

**Pull request merged**